src/activities/Elsa.Activities.Rpa.Web/Services/BrowserFactory.cs

```diff
@@ -0,0 +1,78 @@
+using Elsa.Activities.Rpa.Web.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elsa.Activities.Rpa.Web.Services
+{
+    public class BrowserFactory : IBrowserFactory
+    {
+        private readonly RpaWebOptions _options;
+        private readonly ILogger<BrowserFactory> _logger;
+
+        public BrowserFactory(
+            IOptions<RpaWebOptions> options,
+            ILogger<BrowserFactory> logger
+        )
+        {
+            _options = options.Value;
+            _logger = logger;
+        }
+        private Dictionary<string, IWebDriver> _drivers { get; set; } = new Dictionary<string, IWebDriver>();
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="driverType"><see cref="DriverType"/></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>the driverId</returns>
+        public async Task<string> OpenAsync(string driverType, object? options = default, CancellationToken cancellationToken=default)
+        {
+            var id = Guid.NewGuid().ToString();
+            IWebDriver driver = default;
+            switch(driverType)
+            {
+                case DriverType.Chrome:
+                    {
+                        var chromeDriverInstaller = new ChromeDriverInstaller();
+                        var chromeVersion = await chromeDriverInstaller.GetChromeVersion();
+                        await chromeDriverInstaller.Install(chromeVersion);
+                        //chromeOptions.AddArguments("headless");
+                        driver = new ChromeDriver(options as ChromeOptions);
+                        break;
+                    }
+                case DriverType.Firefox:
+                    {
+                        throw new NotImplementedException();
+                    }
+                case DriverType.InternetExplorer:
+                    {
+                        throw new NotImplementedException();
+                    }
+                case DriverType.Opera:
+                    {
+                        throw new NotImplementedException();
+                    }
```

Comment:
It's OK to leave this in as not implemented, but then I think we also need to make sure that the user can't select these options from the UI.

Reply:
good point, the implementation for those browser is supported by Selenium. The only reason why I've left them not implemented is that I've not yet found a smart way to download appropriate WebDriver as I've done with ChromeDriver instead